mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -1898,6 +1898,13 @@ void ModuleTranslation::setLoopMetadata(Operation *op,
   inst->setMetadata(llvm::LLVMContext::MD_loop, loopMD);
 }
 
+void ModuleTranslation::setDisjointFlag(Operation *op, llvm::Value *inst) {
+  auto iface = cast<DisjointFlagInterface>(op);
+  auto inst_disjoint = cast<llvm::PossiblyDisjointInst>(inst);
```

Comment:
```suggestion
  auto disjointInst = cast<llvm::PossiblyDisjointInst>(inst);
```
nit: missed this before.

Reply:
Done.

Reply:
Can you rename inst_disjoint in the other function as well? Sorry I forgot to mark that as well. Feel free to fix this things proactively.

Regarding the llvm::Value. I think you could use llvm::Instruction and it would cast to a llvm::Value right? That maybe preferable over using a value since it is in line with all the other helper functions we have. If it doesn't work the comment on the function needs to be fixed by replacing inst -> value.

Sorry for being a bit pedantic :).

Reply:
Sorry, I forgot about the other function.

As for the Value issue, I may have been unclear, but the issue I get if I use Instruction is from the llvm builder
```
/home/leon_frenot/fork-llvm/build/tools/mlir/include/mlir/Dialect/LLVMIR/LLVMConversions.inc:279:43: error: invalid conversion from ‘llvm::Value*’ to ‘llvm::Instruction*’ [-fpermissive]
  279 |     moduleTranslation.setDisjointFlag(op, inst);
         |                                                                 ^~~~
         |                                                                 |
         |                                                                 llvm::Value*
```
So the issue comes from builder.CreateOr returning a Value instead of an Instruction.

For now I'll simply fix the comment

Reply:
Done.

Reply:
That makes sense!